### ISSUE TITLE
Update Search is not working in the extended search

### ISSUE THREAD

**Issue by DesBw** [texstudio-org/texstudio#160]
### Environment
- TeXstudio 2.12.8 
- OS: Mac High Sierra 10.13.4
- TeX distribution: Texlive 2018

### Expected behavior

Clicking "update search" is supposed to make a new search (update the search with the new term)

### Actual behavior
It doesn't update the search. It keeps the old term

### How to reproduce

1) open the extended search
2) search one term: "my old term"
3) now, you want to make a modified version of the term: change "my old term" to "my new term"
4) click on "update search". 
The results you get are still for "my old term" not for "my new term"

Note: this used to work in version 2.12.6

**Comment by Dellu** [comment]
I think this is a bug in the latest version. It used work fine in the previous version. 

Completely closing the extended search and opening it updates the result.

**Comment by sunderme** [comment]
maybe the term "update search" is a bit misleading. It update from changes in the combo box which state "current file" "all docs" "project"
To update the search word, you need to press the "+" again in the search panel...

**Comment by Dellu** [comment]
I don't think it is just about terminology. In the earlier versions, I think it used to update the terms themselves, not just the scope.

**Comment by Dellu** [comment]
>"Update Seach" button uses the content of an input field in a totally different part of the GUI
I don't understand what you mean by this. 

Why is the old system used in the version 2.12.6 and before considered outdated?

From the user's perspective, the old system was perfect because you can run multiple searches without leaving the Extended Search window. IN the current system, I have to close and start a new search every time I need to make a new search. For those of us who use the Extended search all day, doing thousands of searches across many files, the new behavior is really hectic. 

A better strategy would be **changing the Scope would always update the search scope in the background**, while the manual "update search" would be reserved for the input field.

**Comment by Dellu** [comment]
Thank you. This is much better.
